misc.py

```diff
@@ -1722,6 +1723,9 @@ def __init__(self):
                 self.psValue = QtWidgets.QSpinBox()
                 self.psValue.setRange(0, 2147483647) # maximum value allowed by qt
 
+                # Puzzle directory
+                self.puzzleDir = QtWidgets.QLineEdit()
```

Comment:
Making this a `QLineEdit` makes it harder to select a folder, and requires additional checks when using this value (i.e. is this a valid path, does the path exist, is it a folder, etc.).